nodejs/eks/nodegroup.ts

```diff
@@ -1197,7 +1197,7 @@ ${customUserData}
                     values: [id],
                 },
             ],
-        }),
+        }, { parent, provider}),
```

Comment:
I suspect the only change necessary to address the issue is this line, which should be:

```suggestion
        }, { parent }),
```

I suspect all the other changes in this PR to take a `provider` are unnecessary (sorry to lead you astray about `provider`). The provider from the parent should be used implicitly, and it looks like in this one place we missed passing `parent` to the `getAmi` call.

@ziye-dm, would you mind giving that a try and letting us know if it addresses the issue for you? Thanks!

Reply:
Yes only one line change could fix my problem and I have tried it's working. See https://github.com/pulumi/pulumi-eks/issues/844#issuecomment-1373357823

I don't really understand how other places work and could you please answer the question that @thomas11 mentioned?
https://github.com/pulumi/pulumi-eks/issues/844#issuecomment-1373781127
> I still wonder about this line (1058). Shouldn't we add `provider there, too?
> const awsRegion = pulumi.output(aws.getRegion({}, { parent, async: true }));

I am ok to change back to one line change if you two are aligned. :D

Reply:
Get it! Only `parent` is needed! It's working!